File: no_nonesense_rust/no-nonsense-c-source-code/02-stdin-stdout/transformer.c
#include <stdio.h>

int encryption_key = 42;

int main()
{
    // char -> 8 bit number (values 0..255)
    char c;
    while ( (c = getc(stdin)) != EOF )
    {
        // write ANY transformation here -> converts ANY input to ANY output
        putc(c ^ encryption_key, stdout);
    }
}

File: no_nonesense_rust/no-nonsense-rust-source-code/02-stdin-stdout/transformer.rs
use std::io::{self, Read, Write};

fn main() {
    let key: u8 = 42;

    // u8 -> 8 bit number (values 0..255)
    for byte in io::stdin().bytes() {
        let b = byte.unwrap();
        // write ANY transformation here -> converts ANY input to ANY output
        io::stdout().write_all(&[b ^ key]).unwrap();
    }
}
